### PR TITLE
[4.0] render modals consistently

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue 
https://github.com/joomla/joomla-cms/pull/21847 .

### Summary of Changes
Modals should be placed before `</body>` tag.

### Testing Instructions
Apply patch, check the batch operation on article list view

### CODE REVIEW

**Comment by ghost** [comment]
> check the batch operation on article list view

what to look for?

**Comment by dgrammatiko** [comment]
@franz-wohlkoenig check the issue: https://github.com/joomla/joomla-cms/issues/19408

**Comment by TobsBobs** [comment]
> Testing Instructions
See here: #21343 you can install the patchtester from here and test the "fetch data" modal.

**Comment by ghost** [comment]
Batch operation on article list view works as expected without this PR.

"Fetch Data" in Patchtester works as expected with this PR.

Thats why i was asking.

**Comment by bembelimen** [comment]
With this changes you break all structures regarding CSS.

Before you could add the modal code into your extension with a specific "wrapper class" to format e.g. the content of all modals.

Now the modal is never into the extension again.

**Comment by ghazal** [comment]
I am a bit at a loss here.
#21847 and this one work alright.
But,
> With this changes you break all structures regarding CSS

@bembelimen 
Where do we have to check ?

**Comment by dgrammatiko** [comment]
> Before you could add the modal code into your extension with a specific "wrapper class" to format e.g. the content of all modals.

You can still do that now. All modals have an id so in your layout css you could target the modal contents through that selector

**Comment by bembelimen** [comment]
Yes and no, if you create modals on the fly, you don't know the ID for the CSS file.

**Comment by dgrammatiko** [comment]
> Yes and no, if you create modals on the fly, you don't know the ID for the CSS file.

That's not a problem. You either have a pattern creating the id (eg article_modal_ + artcleId) or you prepend something random and then you take care css in JS.
BTW this is actually something that we need one way or another as the modals will be `dialog` tags and the polyfill suggests that all elements should be before the closing body tag. So that was the plan from the beginning, if you have code that depends on specific position of the modals **you** need to do the adjustments (this is also the case for couple instances in core as well).

File: libraries/cms/html/bootstrap.php [hunk 1]
```diff
@@ -208,7 +208,7 @@ public static function renderModal($selector = 'modal', $params = array(), $body
 
 		static::$loaded[__METHOD__][$selector] = true;
 
-		return LayoutHelper::render('joomla.modal.main', $layoutData);
+		Factory::getDocument()->modals[] = LayoutHelper::render('joomla.modal.main', $layoutData);
```
Comment (Quy) [inline libraries/cms/html/bootstrap.php:211]: Change lines 192 and 203.

**Comment by fancyFranci** [comment]
I have tested this item :white_check_mark: successfully on 08015636de57044fef09ebbd9fd2f09ad8aa0386<br /><br />With the patch the patchtesters modal is working again.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21852">issues.joomla.org/tracker/joomla-cms/21852</a>.</sub>

**Comment by dgrammatiko** [comment]
#21847 is a better fit for the project

**Pull request closed without merging**